File: calcule.php
<!-- Mouad Garroud -->
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Advanced Calculator</title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
    <div class="content">
        <h1>Advanced Calculator</h1>
        <form method="post">
            <input type="number" name="number1" placeholder="Enter Number 1" required>
            <select name="operator" required>
                <option value="" disabled selected>Select Operator</option>
                <option value="add">Addition (+)</option>
                <option value="sub">Subtraction (-)</option>
                <option value="mul">Multiplication (×)</option>
                <option value="div">Division (÷)</option>
                <option value="mod">Modulo (%)</option>
                <option value="pow">Exponentiation (^)</option>
                <option value="sqrt">Square Root (√)</option>
                <option value="abs">Absolute Value (|x|)</option>
                <option value="and">Bitwise AND (&)</option>
                <option value="or">Bitwise OR (|)</option>
                <option value="xor">Bitwise XOR (^)</option>
                <option value="lshift">Left Shift (<<)</option>
                <option value="rshift">Right Shift (>>)</option>
            </select>
            <input type="number" name="number2" placeholder="Enter Number 2 (if required)">
            <button type="submit">Calculate</button>
        </form>
        <p>
            <?php
            if ($_SERVER['REQUEST_METHOD'] == 'POST') {
                if (isset($_POST['number1'], $_POST['operator'])) {
                    $number1 = (float) $_POST['number1'];
                    $operator = $_POST['operator'];
                    $number2 = isset($_POST['number2']) && $_POST['number2'] !== '' ? (float) $_POST['number2'] : null;
                    $result = '';

                    switch ($operator) {
                        case 'add':
                            $result = $number1 + $number2;
                            break;
                        case 'sub':
                            $result = $number1 - $number2;
                            break;
                        case 'mul':
                            $result = $number1 * $number2;
                            break;
                        case 'div':
                            $result = $number2 == 0 ? 'Cannot divide by zero.' : $number1 / $number2;
                            break;
                        case 'mod':
                            $result = $number2 == 0 ? 'Cannot divide by zero.' : fmod($number1, $number2);
                            break;
                        case 'pow':
                            $result = pow($number1, $number2);
                            break;
                        case 'sqrt':
                            $result = $number1 < 0 ? 'Cannot calculate square root of a negative number.' : sqrt($number1);
                            break;
                        case 'abs':
                            $result = abs($number1);
                            break;
                        case 'and':
                            $result = (int)$number1 & (int)$number2;
                            break;
                        case 'or':
                            $result = (int)$number1 | (int)$number2;
                            break;
                        case 'xor':
                            $result = (int)$number1 ^ (int)$number2;
                            break;
                        case 'lshift':
                            $result = (int)$number1 << (int)$number2;
                            break;
                        case 'rshift':
                            $result = (int)$number1 >> (int)$number2;
                            break;
                        default:
                            $result = 'Invalid operation.';
                    }

                    echo 'The Result is: ' . htmlspecialchars($result);
                }
            }
            ?>
        </p>
    </div>
</body>
</html>

File: calcule.c
//Mouad Garroud
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

int main() {
    int i, x, o;
    printf("First: ");
    scanf("%d", &i);
    
    printf("Choose (1 +) (2 *) (3 /) (4 -) (5 sqrt) (6 abs) (7 %) (8 compare) (9 exp) (10 <<) (11 >>) (12 &) (13 |): ");
    scanf("%d", &x);

    switch (x) {
        case 1:
            printf("Second: ");
            scanf("%d", &o);
            printf("The sum: %d\n", i + o);
            break;
        case 2:
            printf("Second: ");
            scanf("%d", &o);
            printf("The product: %d\n", i * o);
            break;
        case 3:
            printf("Second: ");
            scanf("%d", &o);
            if (o == 0) {
                printf("Not possible because the second number %d is zero\n", o);
            } else {
                printf("The division: %.2f\n", (double)i / o);
            }
            break;
        case 4:
            printf("Second: ");
            scanf("%d", &o);
            printf("The difference: %d\n", i - o);
            break;
        case 5:
            if (i < 0) {
                printf("%d is a negative number, cannot compute square root.\n", i);
            } else {
                printf("The square root of %d is %.2f\n", i, sqrt(i));
            }
            break;
        case 6:
            printf("The absolute value of %d is %d\n", i, abs(i));
            break;
        case 7:
            printf("Second: ");
            scanf("%d", &o);
            if (o <= 0) {
                printf("The second number should be positive\n");
            } else {
                printf("The modulo is: %d\n", i % o);
            }
            break;
        case 8:
            printf("Second: ");
            scanf("%d", &o);
            if (i < o) {
                printf("The second number %d is bigger than the first number %d\n", o, i);
            } else if (i > o) {
                printf("The first number %d is bigger than the second number %d\n", i, o);
            } else {
                printf("The two numbers are equal: %d = %d\n", i, o);
            }
            break;
        case 9:
            printf("Exponent: ");
            scanf("%d", &o);
            printf("The result of %d raised to the power of %d is %.2f\n", i, o, pow(i, o));
            break;
        case 10:
            printf("Second: ");
            scanf("%d", &o);
            printf("The result of left shift: %d\n", i << o);
            break;
        case 11:
            printf("Second: ");
            scanf("%d", &o);
            printf("The result of right shift: %d\n", i >> o);
            break;
        case 12:
            printf("Second: ");
            scanf("%d", &o);
            printf("The result of bitwise AND: %d\n", i & o);
            break;
        case 13:
            printf("Second: ");
            scanf("%d", &o);
            printf("The result of bitwise OR: %d\n", i | o);
            break;
        default:
            printf("Invalid operation\n");
    }

    return 0;
}
